benchmark/opperf/utils/op_registry_utils.py

```diff
@@ -0,0 +1,193 @@
+    argument names, argument types.
+
+    Parameters
+    ----------
+    op_name: str
+        Name of the operator
+
+    op_handle: OpHandle
+        Handle for the operator
+
+    Returns
+    -------
+    (narg, arg_names, arg_types)
+    """
+    real_name = ctypes.c_char_p()
+    desc = ctypes.c_char_p()
+    num_args = mx_uint()
+    arg_names = ctypes.POINTER(ctypes.c_char_p)()
+    arg_types = ctypes.POINTER(ctypes.c_char_p)()
+    arg_descs = ctypes.POINTER(ctypes.c_char_p)()
+    key_var_num_args = ctypes.c_char_p()
+    ret_type = ctypes.c_char_p()
+
+    check_call(_LIB.MXSymbolGetAtomicSymbolInfo(
+        op_handle, ctypes.byref(real_name), ctypes.byref(desc),
+        ctypes.byref(num_args),
+        ctypes.byref(arg_names),
+        ctypes.byref(arg_types),
+        ctypes.byref(arg_descs),
+        ctypes.byref(key_var_num_args),
+        ctypes.byref(ret_type)))
+
+    narg = int(num_args.value)
+    arg_names = [py_str(arg_names[i]) for i in range(narg)]
+    arg_types = [py_str(arg_types[i]) for i in range(narg)]
+
+    return narg, arg_names, arg_types
+
+
+def _set_op_arguments(mx_operators):
+    """Fetch and set operator arguments - nargs, arg_names, arg_types
+    """
+    for op_name in mx_operators:
+        op_handle = _get_op_handles(op_name)
+        narg, arg_names, arg_types = _get_op_arguments(op_name, op_handle)
+        mx_operators[op_name]["params"] = {"narg": narg,
+                                           "arg_names": arg_names,
+                                           "arg_types": arg_types}
+
+
+def _get_all_mxnet_operators():
+    # Step 1 - Get all registered op names
+    mx_operators = _get_all_registered_ops()
+
+    # Step 2 - Get all parameters for the operators
+    _set_op_arguments(mx_operators)
+    return mx_operators
+
+
+def prepare_op_inputs(arg_params, arg_values):
+    # For each default value combination, prepare inputs
+    inputs = []
+    for arg_value in arg_values:
+        inp = {}
+        for arg_name in arg_params["params"]["arg_names"]:
+            inp[arg_name] = arg_value[arg_name]
+        inputs.append(inp)
+    return inputs
+
+
+def get_all_broadcast_binary_operators():
+    """Gets all binary broadcast operators registered with MXNet.
+
+    Returns
+    -------
+    {"operator_name": {"has_backward", "nd_op_handle", "params"}}
+    """
+    # Get all mxnet operators
+    mx_operators = _get_all_mxnet_operators()
+
+    # Filter for binary broadcast operators
+    binary_broadcast_mx_operators = {}
+    for op_name, op_params in mx_operators.items():
+        if op_name.startswith("broadcast_") and op_params["params"]["narg"] == 2 and \
+                "lhs" in op_params["params"]["arg_names"] and \
+                "rhs" in op_params["params"]["arg_names"]:
+            binary_broadcast_mx_operators[op_name] = mx_operators[op_name]
+    return binary_broadcast_mx_operators
+
+
+def get_all_element_wise_binary_operators():
```

Comment:
can we rename to "elemwise" for consistency with the C source files?

Reply:
Done